src/server/zset_family.cc

```diff
@@ -2743,8 +2757,12 @@ void ZSetFamily::GeoDist(CmdArgList args, ConnectionContext* cntx) {
 
 namespace {
 // Search all eight neighbors + self geohash box
-bool MembersOfAllNeighbors(ConnectionContext* cntx, string_view key, const GeoHashRadius& n,
-                           const GeoShape& shape_ref, GeoArray* ga, unsigned long limit) {
+// write_mode:  Command in WRITE mode
+//              True:  Using transaction->Execute()
+//              False: Using transaction->ScheduleSingleHopT()
+OpStatus MembersOfAllNeighbors(ConnectionContext* cntx, string_view key, const GeoHashRadius& n,
+                               const GeoShape& shape_ref, GeoArray* ga, unsigned long limit,
+                               bool write_mode = false) {
```

Comment:
again, changing function behavior using `write_mode` is not a good approach, imho.
Instead, I would break this function into smaller function. For example, factor out the preparation code  into a separate function that just returns `range_specs`.